### PR TITLE
Implement block producers search component

### PR DESCRIPTION
**Pull request by anastasiia-bilova**

### CODE REVIEW

File: src/components/pages/IndexPage.vue [hunk 1]
```diff
@@ -27,7 +44,16 @@ export default {
...
         statusCode: null,
       },
-      blockProducers: null
+      blockProducers: null,
+      blockProducersSearch: null,
+      searchPhrase: null,
+    }
+  },
+  watch: {
+    searchPhrase: function (val) {
```
Comment (dmytrostriletskyi) [inline src/components/pages/IndexPage.vue]: `val` > `searchPhrase` if possible.

File: src/components/pages/IndexPage.vue [hunk 2]
```diff
@@ -39,6 +65,18 @@ export default {
       if (mutation.type === INDEX_PAGE_ADD_BLOCK_PRODUCERS_MUTATION) {
         this.blockProducers = state.indexPage.blockProducers
       }
+
+      if (mutation.type === blockProducerStorageMutations.subscribe.addError) {
+        this.error = state.blockProducer.error
+      }
+
+      if (mutation.type === blockProducerStorageMutations.subscribe.addFieldsErrors) {
```
Comment (dmytrostriletskyi) [inline src/components/pages/IndexPage.vue]: Remove this, please.

File: src/store/modules/blockProducer.js [hunk 3]
```diff
@@ -109,6 +113,9 @@ export const blockProducer = {
     updateReferenceLinks (state, isUpdated) {
       state.isUpdated = isUpdated
     },
+    searchBlockProducers (state, blockProducersSearch) {
+      state.blockProducersSearch = blockProducersSearch
```
Comment (dmytrostriletskyi) [inline src/store/modules/blockProducer.js]: `blockProducersSearch` > `searchedBlockProducers`.

**Review by dmytrostriletskyi — APPROVED** [review]

File: src/components/pages/IndexPage.vue (hunk 2 above)
Comment (anastasiia-bilova) [reply src/components/pages/IndexPage.vue]: @dmytrostriletskyi, done.

File: src/store/modules/blockProducer.js (hunk 3 above)
Comment (anastasiia-bilova) [reply src/store/modules/blockProducer.js]: @dmytrostriletskyi, done.

File: src/components/pages/IndexPage.vue (hunk 1 above)
Comment (anastasiia-bilova) [reply src/components/pages/IndexPage.vue]: @dmytrostriletskyi, done.

**Pull request merged**